hermeto/core/package_managers/javascript/js_utils.py

```
@@ -0,0 +1,181 @@
+    url: str
+    ref: str = ""
+
+    def __init__(self, url: str, ref: str = "") -> None:
+        """Allow ``GitCloneUrl(url, ref)`` positional construction."""
+        try:
+            super().__init__(url=url, ref=ref)
+        except ValidationError as e:
+            raise UnexpectedFormat(
+                f"Cannot parse git URL: {url}",
+                solution="Ensure the git dependency has a valid URL.",
+            ) from e
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)